cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1128 @@
+{
+  constexpr int block_dim        = UnrolledPolicy::BLOCK_THREADS;
+  constexpr int items_per_thread = UnrolledPolicy::ITEMS_PER_THREAD;
+  constexpr int tile_size        = block_dim * items_per_thread;
+  const Offset offset            = static_cast<Offset>(blockIdx.x) * tile_size;
+
+  // TODO(bgruber): we could use load vectorization here
+
+  [&](cuda::std::array<value_t<RandomAccessIteartorIn>, items_per_thread>&&... arrays) {
+  // load items_per_thread elements
+#pragma unroll
+    for (int j = 0; j < items_per_thread; ++j)
+    {
+      const auto idx = offset + (j * block_dim + threadIdx.x);
+      if (idx < num_items)
+      {
+        // TODO(bgruber): replace by fold over comma in C++17
+        int dummy[] = {(arrays[j] = ins[idx], 0)..., 0}; // extra zero to handle empty packs
+        (void) &dummy[0]; // MSVC needs extra strong unused warning supression
+      }
+    }
+    // process items_per_thread elements
+#pragma unroll
+    for (int j = 0; j < items_per_thread; ++j)
+    {
+      const auto idx = offset + (j * block_dim + threadIdx.x);
+      if (idx < num_items)
+      {
+        out[idx] = f(arrays[j]...);
+      }
+    }
+  }(cuda::std::array<value_t<RandomAccessIteartorIn>, items_per_thread>{}...);
+}
+
+// used for both, memcpy_async and ublkcp kernels
+template <int BlockThreads>
+struct async_copy_policy_t
+{
+  static constexpr int BLOCK_THREADS = BlockThreads;
+  // items per tile are determined at runtime. these (inclusive) bounds allow overriding that value via a tuning policy
+  static constexpr int MIN_ITEMS_PER_THREAD = 1;
+  static constexpr int MAX_ITEMS_PER_THREAD = 32;
+};
+
+// TODO(bgruber) cheap copy of ::cuda::std::apply, which requires C++17.
+template <class F, class Tuple, std::size_t... Is>
+_CCCL_DEVICE auto poor_apply_impl(F&& f, Tuple&& t, ::cuda::std::index_sequence<Is...>)
+  -> decltype(::cuda::std::forward<F>(f)(::cuda::std::get<Is>(::cuda::std::forward<Tuple>(t))...))
+{
+  return ::cuda::std::forward<F>(f)(::cuda::std::get<Is>(::cuda::std::forward<Tuple>(t))...);
+}
+
+template <class F, class Tuple>
+_CCCL_DEVICE auto poor_apply(F&& f, Tuple&& t)
+  -> decltype(poor_apply_impl(
+    ::cuda::std::forward<F>(f),
+    ::cuda::std::forward<Tuple>(t),
+    ::cuda::std::make_index_sequence<::cuda::std::tuple_size<::cuda::std::__libcpp_remove_reference_t<Tuple>>::value>{}))
+{
+  return poor_apply_impl(
+    ::cuda::std::forward<F>(f),
+    ::cuda::std::forward<Tuple>(t),
+    ::cuda::std::make_index_sequence<::cuda::std::tuple_size<::cuda::std::__libcpp_remove_reference_t<Tuple>>::value>{});
+}
+
+// mult must be a power of 2
+template <typename Integral>
+_CCCL_HOST_DEVICE inline auto round_up_to_po2_multiple(Integral x, Integral mult) -> Integral
+{
+  assert(::cuda::std::popcount(static_cast<::cuda::std::__make_unsigned_t<Integral>>(mult)) == 1);
+  return (x + mult - 1) & ~(mult - 1);
+}
+
+template <typename T>
+_CCCL_HOST_DEVICE const T* round_down_ptr(const T* ptr, unsigned alignment)
```

Comment:
`inline`?

Reply:
It's already `inline` since it's a template. But I applied `_CCCL_FORCEINLINE` now to this and some related functions, since those are used in performance critical regions.